File: backend/level5/Action.py
class Rental_action:
    def __init__(self,price,commission):
        self.action = Rental_action.find_actions(price,commission)

    #changes the presentation of the json file
    def find_actions(price,commission):
        print(commission)
        actions = [["driver", "debit", price],
        ["owner", "credit", price-sum(commission.values())],
        ["insurance", "credit", commission["insurance_fee"]],
        ["assistance", "credit", commission["assistance_fee"]],
        ["drivy", "credit", commission["drivy_fee"]]]
        keys = ["who","type","amount"]
        action = [dict(zip(keys,i)) for i in actions]
        return action

File: backend/level4/Action.rb
class Rental_action
    attr_reader :actions
    def initialize(price,commission) 
        @price = price
        @commission = commission
        @actions = Rental_action.find_actions(@price.to_i,@commission)
    end
    #changes the presentation of the json file
    def self.find_actions(price,commission)
        #mapping the new values to the new dictionaries format
        actions = [
            ["driver", "debit", price],
            ["owner", "credit", price-commission.sum{|s,t| t}],
            ["insurance", "credit", commission[:insurance_fee]],
            ["assistance", "credit", commission[:assistance_fee]],
            ["drivy", "credit", commission[:drivy_fee]]].map {|who,type,amount| {"who" =>who, "type"=> type, "amount"=> amount}}
        return actions
    end
end